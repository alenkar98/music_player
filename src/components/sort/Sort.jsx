import { FaAngleDown, FaAngleUp } from 'react-icons/fa';
import './sort.css';
import { LuArrowDownUp } from 'react-icons/lu';
import { useState } from 'react';
import { setSongs } from '../../redux/action/songAction';
import { useDispatch, useSelector } from 'react-redux';

export default function Sort() {
  const [isOpen, setIsOpen] = useState(false);
  const [sortType, setSortType] = useState('');

  function handleClick() {
    setIsOpen(!isOpen);
  }

  const dispatch = useDispatch();
  const songs = useSelector((state) => state.song.songs);

  const handleAscending = () => {
    setSortType(' Asc');
    const sortedSongs = [...songs].sort((a, b) => a.songIndex - b.songIndex);
    dispatch(setSongs(sortedSongs));
    setIsOpen(!isOpen);
  };

  const handleDescending = () => {
    setSortType(' Desc');
    const sortedSongs = [...songs].sort((a, b) => b.songIndex - a.songIndex);
    dispatch(setSongs(sortedSongs));
    setIsOpen(!isOpen);
  };

  return (
    <div className={'sort-container'}>
      <button className="sort" onClick={handleClick}>
        <LuArrowDownUp />
        <span>Sort By{sortType}</span>
        {isOpen ? <FaAngleUp /> : <FaAngleDown />}
      </button>
      {isOpen && (
        <div className="dropDown">
          <button onClick={handleAscending}>Ascending</button>
          <button onClick={handleDescending}>Descending</button>
        </div>
      )}
    </div>
  );
}
