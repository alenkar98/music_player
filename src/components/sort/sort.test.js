import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'; // You may need to install this package

import Sort from './Sort';

const mockStore = configureStore([]);

test('Sort component renders correctly', () => {
  const store = mockStore({ song: { songs: [] } }); // Initialize the store with the necessary state
  const component = renderer.create(
    <Provider store={store}>
      <Sort />
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
