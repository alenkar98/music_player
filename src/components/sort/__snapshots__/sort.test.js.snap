// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort component renders correctly 1`] = `
<div
  className="sort-container"
>
  <button
    className="sort"
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m3 16 4 4 4-4"
      />
      <path
        d="M7 20V4"
      />
      <path
        d="m21 8-4-4-4 4"
      />
      <path
        d="M17 4v16"
      />
    </svg>
    <span>
      Sort By
    </span>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 320 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
      />
    </svg>
  </button>
</div>
`;
